packages/aws-cdk-lib/aws-stepfunctions/lib/states/custom-state.ts

```diff
@@ -74,11 +75,64 @@ export class CustomState extends State implements IChainable, INextable {
       ...this.renderRetryCatch(),
     };
 
-    // merge the Retry field defined in the stateJson into the state
+    if (this.hasMultipleRetrySources(state)) {
+      this.addMultipleRetrySourcesWarning();
+    }
+
+    if (this.hasMultipleCatchSources(state)) {
+      this.addMultipleCatchSourcesWarning();
+    }
```

Comment:
We check to see if we have multiple retry/catch sources to add warnings then check again to generate the state. Can we do 1 check, send a warning then generate the stateJson

Reply:
I meant can't we do something like this in line 92
```suggestion
   if (Array.isArray(this.stateJson.Catch)) {
        if(Array.isArray(state.Catch)) {
               this.addMultipleCatchSourcesWarning();
               state.Catch = [...state.Catch, ...this.stateJson.Catch];
        }
        else
        {
               state.Catch = [...this.stateJson.Catch];
        }
    }
```

Reply:
As mentioned, we initialize the state's `catch` field as an empty array, so even if customers do not call `addCatch()` on their custom state they will incorrectly be flagged for providing both sources. The same is true for retries.

Reply:
Talked offline, I was confused by the `if(Array.isArray(state.Catch))` since its always an array but its only there to act as a type check which i believe is fine